### PR TITLE
Add ruby gemspec

### PR DESCRIPTION
**Pull request by will89**
We can't bundle install a specific git commit with avro since the repo has no gemspec. This adds a gemspec for getting bundle install to work with git.

This is not a drop in replacement for the whole avro release pipeline. Skipping the copying of version from shared file and just hard coding it for now.

prime: @jturkel

### CODE REVIEW

File: lang/ruby/avro.gemspec [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+
+Gem::Specification.new do |spec|
+  spec.name          = 'avro'
+  spec.version       = '1.10.1'
```
Comment (jturkel) [inline lang/ruby/avro.gemspec]: Think it's worth reading this value from `VERSION.txt` i.e. `File.read(File.join(__dir__, 'VERSION.txt')).strip` to get us slightly closer to working with the Avro versioning infrastructure?

File: lang/ruby/Rakefile [hunk 2]
```diff
@@ -15,47 +15,12 @@
 # limitations under the License.
 
 require 'rubygems'
-require 'echoe'
```
Comment (jturkel) [inline lang/ruby/Rakefile:18]: Should we comment this out instead and document why we're doing that? Might make it easier to make these changes real in the future. Same comment about the `dist` task.

File: lang/ruby/Rakefile [hunk 3]
```diff
@@ -15,47 +15,12 @@
...
-  p.author = "Apache Software Foundation"
-  p.email = "dev@avro.apache.org"
-  p.summary = "Apache Avro for Ruby"
-  p.description = "Avro is a data serialization and RPC format"
-  p.url = "https://avro.apache.org/"
-  p.runtime_dependencies = ["multi_json ~>1"]
-  p.licenses = ["Apache-2.0"]
-end
 
-t = Rake::TestTask.new(:interop)
```
Comment (jturkel) [inline lang/ruby/Rakefile:31]: Why did we have to remove the interop tests?

**Review by jturkel — COMMENTED** [review]
A few minor comments but awesome you were able to get this working!

File: lang/ruby/avro.gemspec (hunk 1 above)
Comment (will89) [reply lang/ruby/avro.gemspec]: Yeah, good idea. I'll give that a try.

File: lang/ruby/Rakefile (hunk 3 above)
Comment (will89) [reply lang/ruby/Rakefile:31]: When I tried running it, it didn't work. I'll re-add stuff to the Rakefile and comment out where necessary.

File: lang/ruby/Rakefile [hunk 4]
```diff
@@ -20,7 +20,49 @@ require 'bundler/setup'
 
 Bundler::GemHelper.install_tasks
 
+# Deferring this to the gemspec file
```
Comment (jturkel) [inline lang/ruby/Rakefile]:
```suggestion
# Deferring this to the gemspec file so we can bundle the gem directly from Github.
```

**Review by jturkel — APPROVED** [review]

**Pull request merged**